packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2432,6 +2432,7 @@ class TextSelectionGestureDetectorBuilder {
     // vs [TapGestureRecognizer.onSecondaryTapUp] instead of having to track state in
     // renderEditable. When this migration is complete we should remove this hack.
     // See https://github.com/flutter/flutter/issues/115130.
+    // TODO(luanpotter): Provide kind and buttons here once [TapDragDownDetails] has them.
```

Comment:
I think if we want to expose the `*Details.buttons` API it would be preferred if this pull request handled adding them to all the `*Details` objects and not only `TapDownDetails` and `TapUpDetails`. This would be similar to how we added the `allowedButtonsFilter` API https://github.com/flutter/flutter/pull/111852 on all the recognizers in one PR. I think the benefit of doing it this way versus the alternative of multiple PRs is that the framework is not left in a inconsistent state. What do you think about this?